Journal/Journal/Controllers/EntryController.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  EntryController.swift
+//  Journal
+//
+//  Created by Angel Buenrostro on 1/24/19.
+//  Copyright © 2019 Angel Buenrostro. All rights reserved.
+//
+
+import Foundation
+
+class EntryController {
+    
+    var entries: [Entry] = [] // to be used as the datasource for the class
+    
+    static let baseURL: URL = URL(string: "https://journal-angel.firebaseio.com/")!
+    
+    func put(entry : Entry, completion: @escaping (Error?) -> Void){
+        var url = EntryController.baseURL
+        url = url.appendingPathComponent(entry.identifier)
+        url = url.appendingPathExtension("json")
+        
+        var request = URLRequest(url: url)
+        request.httpMethod = "PUT"
+        
+        do {
+            let encoder = JSONEncoder()
+            request.httpBody = try encoder.encode(entry)
+        } catch {
+            print(error)
+            completion(error)
+            return
+        }
+        
+        URLSession.shared.dataTask(with: request) { (_, _, error) in
+            if let error = error {
+                print(error)
+                completion(error)
+                return
+            }
+            completion(nil)
+        }.resume()
+    }
+    
+    
+    func createEntry(title: String, bodyText: String, completion: @escaping (Error?) -> Void) {
+        let entry = Entry(title: title, bodyText: bodyText)
+        entries.append(entry)
+        
+        put(entry: entry) { (completion) in
+        }
+    }
+    
+    func update(entry: Entry, title : String, bodyText : String, completion: @escaping (Error?) -> Void) {
+        let oldEntry = entry
+        var index = 0
+        while index < entries.count {
+            if entries[index] == oldEntry {
+                entries[index].title = title
+                entries[index].bodyText = bodyText
+                put(entry: entry, completion: completion)
```

Comment:
Entry is a struct, a value type. So you're changing the entry's values in the array but that doesn't affect the entry which is you passed as an argument to the function. Then you're sending the entry which still has old value to the server again. So you're updating an entry in the array but sending unmodified entry to the server
```suggestion
               let updatedEntry = entries[index]
                put(entry: updatedEntry, completion: completion)
```